File: src/api/v1/metrics.rs
// SPDX-License-Identifier: EUPL-1.2
// SPDX-FileCopyrightText: OpenTalk Team <mail@opentalk.eu>

use axum::routing::get;

use crate::api::{Context, Router};
use axum::extract::State;

pub(crate) async fn metrics(_context: State<Context>) -> &'static str {
    "very cool metrics"
}

pub(crate) fn routes() -> Router {
    Router::new().route("/metrics", get(metrics))
}

File: load-test/src/lib/metrics.js
// SPDX-License-Identifier: EUPL-1.2
// SPDX-FileCopyrightText: OpenTalk Team <mail@opentalk.eu>

/**
 * @param {string} testName - Name of the test
 * @param {Date | string} startTime - Start time of the test, used to differentiate multiple runs
 * @returns {Object} Tags object for metrics
 */
export function shared_tags(testName, startTime) {
  return {
    vu: `${__VU}`,
    iteration: `${__ITER}`,
    ...test_name_tag(testName, startTime),
  };
}

/**
 * @param {string} testName - Name of the test
 * @param {Date | string} startTime - Start time of the test, used to differentiate multiple runs
 * @returns {Object} Tag object with only the test name
 */
export function test_name_tag(testName, startTime) {
  const timeString = startTime instanceof Date ? startTime.toTimeString() : startTime;
  return {
    test_name: `${testName}-${timeString}`,
  };
}
